oeps/oep-0047-bp-semantic-versioning.rst

```diff
@@ -0,0 +1,90 @@
+###########################
+OEP-47: Semantic Versioning
+###########################
+
+.. list-table::
+
+   * - OEP
+     - :doc:`OEP-0047 </oeps/oep-0047>`
+   * - Title
+     - Semantic Versioning
+   * - Last Modified
+     - 2020-05-08
+   * - Authors
+     - Robert Raposa <rraposa@edx.org>
+   * - Arbiter
+     - Régis Behmo <regis@behmo.com>
+   * - Status
+     - Under Review
+   * - Type
+     - Best Practice
+   * - Created
+     - 2020-03-19
+   * - Review Period
+     - 2020-05-11 - 2020-05-25
+
+.. contents::
+   :local:
+   :depth: 3
+
+Summary
+=======
+
+For any Open edX repository that publishes versioned packages, this OEP recommends following the `Semantic Versioning v2`_.
+
+The most relevant part from the summary of the `Semantic Versioning v2`_ specification:
+
+    Given a version number MAJOR.MINOR.PATCH, increment the:
+
+    * *MAJOR* version when you make *incompatible API changes*,
+    * *MINOR* version when you *add functionality in a backwards compatible manner*, and
+    * *PATCH* version when you make *backwards compatible bug fixes*.
+
+Some additional tips are called out in `Backward Incompatible Changes`_ and `Tooling and Automation`_.
```

Comment:
semver v2 does not explicitly state if you should use pre-releases or not and I'm not sure if we currently use them in our libraries.  Is that something we want to provide guidance on?
https://www.python.org/dev/peps/pep-0440/#pre-releases

Reply:
I haven't heard of us doing this.  The intent of this OEP is not to introduce terms or process that is unfamiliar and not already used, but simply to be the "semantic" back into "semantic versioning".  I mostly want a simple OEP to clear up MAJOR vs MINOR vs PATCH.